// create/lib/producers/produceTemplate.js
import { createSystemContextWithAuth } from "../system/createSystemContextWithAuth.js";
import { getPresetByName } from "../utils/getPresetByName.js";
import { applyBlockModifications } from "./applyBlockModifications.js";
import { produceBlocks } from "./produceBlocks.js";
export async function produceTemplate(template, { addons, blocks: blockModifications, directory = ".", mode, offline, options, preset: requestedPreset, ...providedSystem }) {
    const system = await createSystemContextWithAuth({
        directory,
        offline,
        ...providedSystem,
    });
    const preset = typeof requestedPreset === "string"
        ? getPresetByName(template.presets, requestedPreset)
        : requestedPreset;
    if (preset instanceof Error) {
        throw preset;
    }
    const blocks = applyBlockModifications(preset.blocks, blockModifications);
    const creation = produceBlocks(blocks, {
        addons,
        mode,
        offline,
        options,
        system: { ...system, directory },
    });
    return {
        addons: [],
        files: {},
        requests: [],
        scripts: [],
        suggestions: [],
        ...creation,
    };
}
